overlord/ifacestate/handlers.go

```diff
@@ -332,6 +332,18 @@ func (m *InterfaceManager) undoDiscardConns(task *state.Task, _ *tomb.Tomb) erro
 	return nil
 }
 
+func getTaskHookAttributes(task *state.Task) (attrs interfaces.InterfaceAttrs, err error) {
+	attrs.PlugAttrs = make(map[string]interface{})
+	attrs.SlotAttrs = make(map[string]interface{})
+	if err = task.Get("plug-attrs", &attrs.PlugAttrs); err == nil {
```

Comment:
does plug-attrs contains the original attrs from the yaml + what the hook changed? or just new things? the extra  in the arg name to Validate* would make think the latter, but if that's the case the connection check is wrong, that one needs the updated full set or we need to tweak the code in policy a bit differently